src/OpenTelemetry/Metrics/CompositeMetricReaderExt.cs

```diff
@@ -11,126 +10,33 @@ namespace OpenTelemetry.Metrics;
 /// </summary>
 internal sealed partial class CompositeMetricReader
 {
-    internal List<Metric?> AddMetricsWithNoViews(Instrument instrument)
+    internal override List<Metric> AddMetricWithNoViews(Instrument instrument)
     {
-        var metrics = new List<Metric?>(this.count);
+        var metrics = new List<Metric>(this.count);
 
         for (var cur = this.Head; cur != null; cur = cur.Next)
         {
-            metrics.Add(cur.Value.AddMetricWithNoViews(instrument));
-        }
-
-        return metrics;
-    }
-
-    internal void RecordSingleStreamLongMeasurements(List<Metric?> metrics, long value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-    {
-        Debug.Assert(metrics.Count == this.count, "The count of metrics to be updated for a CompositeReader must match the number of individual readers.");
-
-        int index = 0;
-        for (var cur = this.Head; cur != null; cur = cur.Next)
-        {
-            var metric = metrics[index];
-            if (metric != null)
-            {
-                cur.Value.RecordSingleStreamLongMeasurement(metric, value, tags);
-            }
-
-            index++;
-        }
-    }
-
-    internal void RecordSingleStreamDoubleMeasurements(List<Metric?> metrics, double value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-    {
-        Debug.Assert(metrics.Count == this.count, "The count of metrics to be updated for a CompositeReader must match the number of individual readers.");
-
-        int index = 0;
-        for (var cur = this.Head; cur != null; cur = cur.Next)
-        {
-            var metric = metrics[index];
-            if (metric != null)
+            var innerMetrics = cur.Value.AddMetricWithNoViews(instrument);
+            if (innerMetrics.Count > 0)
```

Comment:
Add a `Debug.Assert` here to check that `innerMetrics.Count <= 1`.

Reply:
I pushed this and a few other asserts that seemed to be missing.